lib/common/file-system.ts

```diff
@@ -205,7 +207,13 @@ export class FileSystem implements IFileSystem {
 			space = this.getIndentationCharacter(filename);
 		}
 
-		return this.writeFile(filename, JSON.stringify(data, null, space), encoding);
+		let stringifiedData = JSON.stringify(data, null, space);
```

Comment:
Maybe this should be moved inside an `else` statement. No need to stringify data multiple times.
```TypeScript
let stringifiedData;
if (path.basename(filename) === "package.json") {
	const newline = detectNewline(stringifiedData);
	stringifiedData = stringifyPackage(data, space, newline);
} else {
	stringifiedData = JSON.stringify(data, null, space);
}
```